cassiopeia/tools/branch_length_estimator/IIDExponentialBayesian.py

```diff
@@ -0,0 +1,805 @@
+                    and child_state == tree.missing_state_indicator
+                )
+            self._Ks[child] = k
+        # Check monotonicity of Ks
+        for (parent, child) in tree.edges:
+            assert self._Ks[parent] >= self._Ks[child]
+
+    @property
+    def log_likelihood(self):
+        """
+        The log-likelihood of the training data under the estimated model.
+        """
+        return self._log_likelihood
+
+    def log_joints(self, node: str) -> np.array:
+        """
+        log joint node time probabilities.
+
+        The log joint probability density of the observed tree topology, state
+        vectors, and all possible times for the node. In other words:
+        log P(node time = t, character states, tree topology) for t in [0, T]
+        where T is the discretization_level.
+
+        Args:
+            node: An internal node of the tree, for which to compute the
+                posterior log joint.
+
+        Returns:
+            log P(node time = t, character states, tree topology) for t in
+            [0, T], where T is the discretization_level.
+        """
+        if node not in self._posteriors.keys():
+            raise ValueError(f"Unexistent node: {node}")
+
+        return self._log_joints[node].copy()
+
+    def posterior_time(self, node: str) -> np.array:
+        """
+        The posterior distribution of the time for a node.
+
+        The posterior time distribution of a node, numerically computed, i.e.:
+        P(node time = t | character states, tree topology) for t in [0, T]
+        where T is the discretization_level.
+
+        Args:
+            node: An internal node of the CassiopeiaTree.
+
+        Returns:
+            P(node time = t | character states, tree topology) for t in [0, T]
+            where T is the discretization_level.
+
+        Raises:
+            ValueError if the node is not an internal node.
+        """
+        if node not in self._posteriors.keys():
+            raise ValueError(f"Unexistent node: {node}")
+
+        return self._posteriors[node].copy()
+
+    @property
+    def mutation_rate(self):
+        """
+        The CRISPR/Cas9 mutation rate.
+        """
+        return self._mutation_rate
+
+    @property
+    def birth_rate(self):
+        """
+        The phylogeny birth rate.
+        """
+        return self._birth_rate
+
+    @property
+    def discretization_level(self):
+        """
+        How many timesteps are used to discretize time.
+        """
+        return self._discretization_level
+
+    @property
+    def sampling_probability(self):
+        """
+        The probability that a leaf in the ground truth tree was sampled.
+        """
+        return self._sampling_probability
+
+    @classmethod
+    def exact_log_full_joint(
+        self,
+        tree: CassiopeiaTree,
+        mutation_rate: float,
+        birth_rate: float,
+        sampling_probability: float,
+    ) -> float:
```

Comment:
Yes, you are correct.

I'm thinking about making this a private method, since it is only really useful for testing. A user should never want to use this.

Reply:
I have removed these classmethods completely, opting to place them as functions in the test file.